### PR TITLE
Update wasm handling of embedded bitcode

### PR DESCRIPTION
**Pull request by alexcrichton**
These patches should allow us to remove all wasm-specific behavior from https://github.com/rust-lang/rust/pull/70458, yay!

### CODE REVIEW

File: llvm/lib/CodeGen/TargetLoweringObjectFileImpl.cpp [hunk 1]
```diff
@@ -1688,6 +1688,13 @@ static const Comdat *getWasmComdat(const GlobalValue *GV) {
 }
 
 static SectionKind getWasmKindForNamedSection(StringRef Name, SectionKind K) {
+  // segments within the data section.
+  // This could be avoided if all data segements (the wasm sense) were
+  // represented as thier own sections (in the llvm sense).
```
Comment (bjorn3) [inline llvm/lib/CodeGen/TargetLoweringObjectFileImpl.cpp:1694]: *their

Comment (cuviper) [reply llvm/lib/CodeGen/TargetLoweringObjectFileImpl.cpp:1694]: I went ahead and fixed that upstream: https://github.com/llvm/llvm-project/commit/5a0d8c31a32711dd1101ebcd0a563dec13efdfb9

File: llvm/lib/CodeGen/TargetLoweringObjectFileImpl.cpp [hunk 2]
```diff
@@ -1688,6 +1688,13 @@ static const Comdat *getWasmComdat(const GlobalValue *GV) {
 }
 
 static SectionKind getWasmKindForNamedSection(StringRef Name, SectionKind K) {
+  // segments within the data section.
```
Comment (cuviper) [inline llvm/lib/CodeGen/TargetLoweringObjectFileImpl.cpp:1692]: Hmm, somehow you chopped the first line of this comment, but it's there on #49...

File: llvm/lib/CodeGen/TargetLoweringObjectFileImpl.cpp (hunk 1 above)
Comment (alexcrichton) [reply llvm/lib/CodeGen/TargetLoweringObjectFileImpl.cpp:1694]: Yeah I'm not authoring this code, I'm just cherry-picking from upstream.

File: llvm/lib/CodeGen/TargetLoweringObjectFileImpl.cpp (hunk 2 above)
Comment (alexcrichton) [reply llvm/lib/CodeGen/TargetLoweringObjectFileImpl.cpp:1692]: Oh oops, this didn't merge cleanly in both cases and it looks like I botched this merged but didn't botch the other. Should be fixed now!

**Comment by cuviper** [comment]
LGTM!

**Pull request merged**